### DISCUSSION TITLE
Micro-Swiss nozzle offset affecting bed levelling and build area

### DISCUSSION THREAD

**Discussion by hayhay7789** [Jyers/Marlin#739] [Q&A] [answered]
So I installed a Micro-Swiss direct drive extruder + full metal hotend on my printer, which has been working great so far, except for one thing causing me some grief, which is the extra nozzle offset. Micro-Swiss instructs users to add a negative offset to X_MIN_POS & Y_MIN_POS in the Configuration.h file:

```
#define X_MIN_POS-5
#define Y_MIN_POS-10
```

This initially caused me some trouble with the bed size. Prior to installing direct drive I had my firmware set up to use all 235x235x250mm available on the E3V2 print bed. After installing, I kept the print bed settings as they were, and my settings looked like this:

```
#define X_BED_SIZE 235
#define Y_BED_SIZE 235
// Travel limits (mm) after homing, corresponding to endstop positions.
#define X_MIN_POS -5
#define Y_MIN_POS -10
#define X_MAX_POS X_BED_SIZE + X_MIN_POS
#define Y_MAX_POS Y_BED_SIZE + Y_MIN_POS
```
However, this caused problems with my stepper motors, especially the Y motor, Since its position at the endstop is -10 rather than 0, it would only have counted 225mm when it has already extended to its full limit of 235. Whenever homing the printer or manually moving it to -10 or 235, the bed would extend to its limit but the motor would keep pushing, causing an awful rumbling noise. To fix this, I have set the bed size to 230 and 225. Is this the proper solution to the problem, or is there a better way to correct this without losing bed size?

In addition to that problem, I am also having difficulty levelling the bed properly, as the manual levelling points accessed through Prepare>Manual Levelling are no longer situated over the levelling springs. This makes it pretty difficult to effectively level my bed, and I believe this is causing the very uneven first layers I have been experiencing recently, with the top right too squished and the bottom right so high it's not even adhering to the bed. I am using a BLTouch and have UBL enabled but according to my mesh it's pretty far off. I did some reading and apparently those manual levelling points can be changed in Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp, lines 1257-1309, but I haven't a clue what to change the numbers in there to, nor how to do it, since when I tried it failed to compile.

Hoping someone has some tips. Thanks :)

**Comment by Jyers** [comment, marked as answer]
As far as the endstops are concerned, it might just be a physical limitation. If you move the z axis as far as it can go towards the front of the printer and notice the nozzle is still not off the bed, than you are going to have to sacrifice some print area in the config. If not we'll have to look into what in your config is causing the print head to move too far. 

The manual leveling points are based off your print bed size, so once you have that configured correctly you should be fine. As long as the printer isn't loosing to much travel in either direction, the points should still be approximately over the screws.

**Reply by hayhay7789** [reply]
Gotcha. I have been encountering some quirks with my config which suggests I probably overlooked some stuff (like when I tried enabing UBL with my BLTouch and failed for a couple days before I realized BLtouch wasn't defined in the config). Perhaps I have overlooked or mistakenly changed a setting which would cause these troubles? The settings I put in the OP work okay, but my BLtouch probe misses a pretty significant width from the top and the right as my print bed ends up off center to the bottom left, which is causing my troubles with the manual levelling points defined in the lcd/extui/lib/creality_dwin/creality_dwin.cpp file

**Comment by smooreace** [comment]
I had some issues with my first run of this firmware as well.. more related
to my prints being offset to the front left.... also using a MS hot-end. I
ended up getting good enough results with the setup below that I was happy
enough to stop fighting it
[image: image.png]

**Reply by hayhay7789** [reply]
I believe that is the same problem I am having. My BLTouch misses a significant chunk of the right and top of my bed but prints pretty much right on the edge otherwise so it would seem the printer orients the bed as beginning from the origin point

**Comment by smooreace** [comment]
Just realized that replying via email doesnt properly attach the image


![image](https://user-images.githubusercontent.com/35540547/117912632-752a8180-b2a5-11eb-91b0-db7cccb703c9.png)

**Reply by hayhay7789** [reply]
Thanks for the answer, sorry for the late replies. I gave your settings a shot hoping for a miracle, and unfortunately it behaves weirdly on my printer. The x-carriage doesn't even make it all the way to the x endstop when I manually move the X axis in Control, and the Y axis overtravels in both directions creating that awful noise, despite 230 + -4 being 234, 1mm less than the max travel I am told these printers have.
